File: backend/SyncWire.API/Models/Notification.cs
namespace SyncWire.API.Models;

public class Notification
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public string Title { get; set; } = string.Empty;
    public string Body { get; set; } = string.Empty;
    public string? Channel { get; set; }
    public string? DeviceId { get; set; }
    public DateTime Timestamp { get; set; } = DateTime.UtcNow;
    public NotificationPriority Priority { get; set; } = NotificationPriority.Default;
}

public enum NotificationPriority
{
    Low,
    Default,
    High
}

public class SendNotificationRequest
{
    public string Title { get; set; } = string.Empty;
    public string Body { get; set; } = string.Empty;
    public string? Channel { get; set; }
    public string? DeviceId { get; set; }
    public NotificationPriority Priority { get; set; } = NotificationPriority.Default;
}

File: frontend/src/types/notification.ts
export interface Notification {
  id: string;
  title: string;
  body: string;
  channel?: string;
  deviceId?: string;
  timestamp: string;
  priority: "Low" | "Default" | "High";
}

export interface SendNotificationRequest {
  title: string;
  body: string;
  channel?: string;
  deviceId?: string;
  priority: "Low" | "Default" | "High";
}
